packages/repository-json-schema/src/build-schema.ts

```diff
@@ -15,7 +15,8 @@ import {
 import {JSONSchema6 as JSONSchema} from 'json-schema';
 import {JSON_SCHEMA_KEY, MODEL_TYPE_KEYS} from './keys';
 
-export interface JsonSchemaOptions {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface JsonSchemaOptions<T extends object = any> {
```

Comment:
Let's discuss the default type value. When we use `any`, any code that's not providing `T` is basically throwing away all type checks and there is no warning/error about that problem reported by the compiler.

I am proposing to not provide any default value for `T`, in which case TypeScript will pick `object` as the default. Alternatively, we can also specify `{}` as the default type. To be honest, I am not sure what's the different between `object` and `{}` and cannot tell what is a better option to use.

My expectation is that when `T` is set to `object` or `{}`:
- `keyof T` is resolved as `never`
- an attempt to set `exclude` to a non-empty array fails with a compilation error
- the developer is forced to specify T value for `JsonSchemaOptions`

I see that `export function buildModelCacheKey<T extends object>` already does not provide any default value for `T`, thus falling back to `object` as the default. Since that seems to work well, let's do the same for JsonSchemaOptions too.

```suggestion
export interface JsonSchemaOptions<T extends object> {
```